packages/smart-wallet/src/walletFactory.js

```diff
@@ -112,35 +124,38 @@ export const start = async (zcf, privateArgs) => {
   const creatorFacet = makeHeapFarInstance(
     'walletFactoryCreator',
     M.interface('walletFactoryCreatorI', {
-      provideSmartWallet: M.call(
+      provideSmartWallet: M.callWhen(
         M.string(),
-        M.eref(M.any()),
-        M.eref(M.any()),
-      ).returns(M.promise()),
+        M.remotable(),
+        M.remotable(),
+      ).returns(M.number()),
     }),
     {
       /**
        * @param {string} address
        * @param {ERef<import('@agoric/vats/src/vat-bank').Bank>} bank
-       * @param {ERef<MyAddressNameAdmin>} myAddressNameAdmin
-       * @returns {Promise<import('./smartWallet').SmartWallet>}
+       * @param {ERef<NameAdmin>} namesByAddressAdmin
+       * @returns {Promise<[import('./smartWallet').SmartWallet, boolean]>} wallet
+       *   along with a flag to distinguish between looking up an existing wallet
+       *   and creating a new one.
        */
-      provideSmartWallet(address, bank, myAddressNameAdmin) {
+      provideSmartWallet(address, bank, namesByAddressAdmin) {
+        let created = false;
```

Comment:
this term is a little confusing as every smart wallet is created at some point. 

consider `providerCreated` or the inverse `providerReused`.

Reply:
how about `makerCalled`?